internal/packages/changelog/yaml.go

```diff
@@ -0,0 +1,160 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package changelog
+
+import (
+	"bytes"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// PatchYAML looks for the proper place to add the new revision in the changelog,
+// trying to conserve original format and comments.
+func PatchYAML(d []byte, patch Revision) ([]byte, error) {
+	var nodes []yaml.Node
```

Comment:
I understand why you did it this way (generic YAML parsing), but maybe it's fair to depend on schema and unmarshal the changelog to the proper structure? I don't mind if we lose some comments in the meantime.

Reply:
I would prefer to change the files the minimum possible and maintain the current format, at the end these files are still maintained by humans.

I gave a try to yaml patch libraries for this, and they work well for simple changes as setting the version in the manifest, or appending elements, but I didn't manage to add the changelog entry on top. I can give this approach another try so we don't have to maintain this parsing code, but we can still keep the format and comments.

Reply:
I guess that it's a matter of personal preference, YAML patching vs strict model. I don't have any strong points for switching to mapping except for simpler implementation and less node-level processing.

BTW as an alternative approach, you can consider [mapstr](https://github.com/elastic/elastic-package/blob/main/internal/common/mapstr.go). It's up to you to decide which option seems less error-prone :)

Reply:
Yeah, this code is a bit too much for what it does, but I didn't find another reliable way to do it with the requisite (maybe self-imposed :) ) of maintaining the format as much as possible.

In any case I have tried to maintain the interfaces of the two exposed methods clean, and tests cover them, we can change their implementation if we get another idea.